### ISSUE TITLE
[zfb Migration Parity][Epic] Phase B-16 verify + Phase C/E routing

### ISSUE THREAD

**Issue by Takazudo** [zudolab/zudo-doc#1121]
- https://github.com/zudolab/zudo-doc/issues/663
- Supersedes: #1120

---

**Super-epic:** #663
**Parent (Phase B):** #665
**Upstream blocker resolved:** [Takazudo/zudo-front-builder#95](https://github.com/Takazudo/zudo-front-builder/issues/95) — closed via [zfb PR#96](https://github.com/Takazudo/zudo-front-builder/pull/96) (S0–S6 prod-asset-graph merged into zfb main on 2026-04-30)
**This epic's base branch:** `base/zfb-migration-parity-phase-b16-verify-route`
**Parent branch (PR target):** `base/zfb-migration-parity` (super-epic anchor — NOT `main`)

## Overview

Verification + routing pass after upstream zfb#95 shipped. Three outcomes:

1. **Phase B-16 #1119 closes.** Rebuild local zfb release binary (current `target/release/zfb` predates the zfb#95 merge), refresh the `node_modules/.bin/zfb` symlink, rebuild zudo-doc, and confirm `dist/assets/styles-*.css` plus the `<link rel="stylesheet">` tag in `<head>` are now emitted. Then re-run the parity harness and confirm `asset-loss` drops from 167 → ≈0.

2. **Phase C #666 routing decision.** Triage the surviving 33 `content-loss` routes — 21 of 33 are `/ja/docs/*`, suggesting a possible systematic ja-rendering cluster. Decide: file Phase B-17 sibling epic (shared cause) OR resume Phase C #666 (genuine per-page residuals).

3. **Phase E #668 re-baseline.** Update Phase E's body with the fresh post-B-16 artifact diff numbers and a Phase D / Phase E naming clarification (resume issue #1120 incorrectly called this work "Phase D"; per super-epic #663 the canonical taxonomy is Phase D = symmetric-diff route audit (#667, MERGED via #689) and Phase E = artifact diff verification (#668, OPEN)).

## Sub-issues

| # | Title | Depends on |
|---|---|---|
| #1122 | [Sub] Rebuild zfb, verify B-16 closure, capture artifact baseline | (none — first) |
| #1123 | [Sub] Triage 33 content-loss residuals → Phase B-17 epic OR Phase C resume | #1122 (Branch A) |
| #1124 | [Sub] Re-baseline Phase E #668 with post-B-16 artifact diff | #1122 (Branch A) — runs in parallel with #1123 |
| #1125 | [Sub] Finalize: super-epic comment + close resume issue #1120 | #1123 + #1124 |

## Sub-task dependency graph

```
#1122 (rebuild + verify + capture artifact baseline)
  ├─ Branch A → continue to #1123 + #1124 in parallel
  ├─ Branch B → halt; file harness follow-up OR re-open zfb#95
  └─ Branch C → halt; re-open zfb#95

#1122 (Branch A) ─┬─→ #1123 (triage 33 content-loss → B-17 OR #666)
                  └─→ #1124 (re-baseline #668)
                         │
#1123 ───────────────────┴─→ #1125 (super-epic comment + close #1120)
#1124 ────────────────────────┘
```

## Definition of done

Per #1120's DOD:

- [ ] **Phase B-16 #1119 is closed (asset-loss=0 confirmed)** — covered by #1122 Branch A. Literal target is `asset-loss=0`; ≤5 documented per-route residual is an acceptable fallback ONLY with each route's reason recorded in the closure comment.
- [ ] **Phase C triage decision is made (B-17 epic filed OR Phase C #666 in execution)** — covered by #1123 (both branches).
- [ ] **Phase D started or filed** — explicitly satisfied as follows: canonical **Phase D #667 was already merged** via PR #689 on 2026-04-29 (96 symmetric-difference routes audited, zero accidental losses found, decision doc landed). The work #1120's body called "Phase D" is, per super-epic #663's table, canonically **Phase E #668** (non-HTML artifact diff verification). #1124 re-baselines #668 with the post-B-16 numbers and #1125 records the naming clarification in the close comments. **No new Phase D epic is filed because Phase D is already done.**

## Plan log

`/home/takazudo/cclogs/zudo-doc2/20260501_184500-big-plan-phase-b16-verify-route.md` (local; full plan including review notes from `/codex-2nd` and `/gcoc-2nd`).

## Instructions for `/x-wt-teams`

Run this epic in a fresh Claude Code session:

    /x-wt-teams https://github.com/zudolab/zudo-doc/issues/1121

The session will create `base/zfb-migration-parity-phase-b16-verify-route` off `base/zfb-migration-parity` and merge the resulting epic-PR into the super-epic anchor.

> **Branch hint:** the parent branch is `base/zfb-migration-parity`, not `main`. `/x-wt-teams` should detect this from the user's checkout state but the explicit note here prevents accidental main-targeting.

**Comment by Takazudo** [comment]
🤖 Starting work on this epic in a Claude Code session (`/x-wt-teams` Super-Epic child, Wave 1 of 3).

**Wave plan** (sub-issues have hard dependencies, incompatible with x-wt-teams parallel topics — running as accumulating waves):

- Wave 1 (this session): #1122 only. Epic-PR created and stays open as draft.
- Wave 2 (next session, `--stay`): #1123 + #1124 in parallel.
- Wave 3 (final session, `--stay`): #1125 — performs mandatory epic-PR merge into super-epic base.

Flags forwarded to all waves: `-so -gcoc`. To avoid conflicts, please check the latest comments before starting another session on this epic.

**Comment by Takazudo** [comment]
## Wave 1 complete — Branch B (comparator mismatch); waves 2 + 3 halted

**Epic-PR:** https://github.com/zudolab/zudo-doc/pull/1126 (stays draft across waves)

### What landed (commit `f6857aa`)

1. Rebuilt local zfb with PR#95/96 + #94 + #97; refreshed `node_modules/.bin/zfb` via `pnpm zfb:link`.
2. `scripts/zfb-link.mjs`: added `findTailwindBinary()` to auto-discover the staged `tailwindcss-v4` binary and inline an absolute `ZFB_TAILWIND_BIN` into the wrapper. Same shell pattern as the existing `ZFB_ESBUILD_BIN` inlining; eliminates the manual export step.
3. Captured `docs/migration-check-reports/2026-05-01-round-11-b16-verify.md` (~862 lines, durable evidence).
4. `pnpm build` confirmed: `dist/assets/styles-a0c3a8de.css` emitted; `dist/index.html` `<head>` carries `<link rel="stylesheet" href="/assets/styles-a0c3a8de.css">`. PR#94 + PR#97 spot-checks pass.

The literal output gap that #1119 was filed for is **closed at the build-output level**. zfb#95 is real and works.

### Why Branch B and not Branch A

Round-11 harness reported `asset-loss = 168/221`. Inspection of 5 representative findings showed the dist HTML actually contains the expected stylesheet asset under the new `/assets/styles-*.css` URL shape, but the comparator's `hasAssetLoss()` does literal URL matching against Astro's `/_astro/base.*.css` and treats the framework-equivalent zfb stylesheet as a loss.

So per the #1122 Branch B classification rule: **comparator is the stale party, not zfb**.

### Branch B actions taken

- Filed sibling issue **#1327** — comparator fix work.
- **#1119 stays OPEN** until #1327 ships and a round-12 run shows `asset-loss = 0`.
- **Did NOT re-open `Takazudo/zudo-front-builder#95`** (upstream fix is real).
- **Waves 2 + 3 halted** (#1123, #1124 depend on #1122 Branch A; #1125 depends on those).

### CI

Pre-existing red shape — `zfb: not found` because CI can't build the zfb Rust sibling repo on demand. Same as every prior run on `base/zfb-migration-parity`. Not introduced by this wave.

### Tailwindcss v4 binary

Wave 1's subagent had to download `tailwindcss` v4.2.0 from `tailwindlabs/tailwindcss` releases and stage it at `/home/takazudo/repos/myoss/zfb/crates/zfb/binaries/tailwindcss-v4` to unblock `pnpm build`. The new `scripts/zfb-link.mjs` discovery handles the post-stage developer experience, but the download itself is a one-time manual prerequisite that should probably be documented in zfb's CLAUDE.md or a prerequisite checklist.

### Review

`/gcoc-review` returned surface-level concerns on `scripts/zfb-link.mjs` (bash quoting, path traversal, error handling) that don't apply — paths are derived from controlled local state, the `${VAR:=DEFAULT}` pattern already in use for `ZFB_ESBUILD_BIN`, and the discovery returns `null` cleanly when the binary isn't staged. No fixes applied.

### Next step (out of band)

Comparator fix work belongs on **#1327**, not on this epic-PR. Once #1327 ships and round-12 confirms `asset-loss = 0`, resume #1122 here to close #1119, then waves 2 + 3 unblock.

Recommended next session:

    /x-wt-teams -so -gcoc https://github.com/zudolab/zudo-doc/issues/1327

Local session log: `/home/takazudo/cclogs/zudolab-zudo-doc/20260501_053445-x-wt-teams-phase-b16-verify-route-wave1.md`

**Comment by Takazudo** [comment]
## Epic complete — closing

Re-checked epic state in a fresh `/x-wt-teams` session. All Definition-of-Done items are satisfied; the epic-PR has been merged; no further work belongs here.

### Outcomes (all sub-issues CLOSED)

| Sub | Goal | Outcome |
|---|---|---|
| #1122 | Rebuild zfb, verify B-16 closure, capture artifact baseline | DONE via #1327 session — `pnpm migration-check --rerun` after the comparator fix dropped asset-loss 167/219 → 2/221 (≤5 documented residual: framework path-format differences for Astro Optimize Image hashed `.webp` vs zfb's unhashed `/img/...`). Round-12 evidence committed. |
| #1123 | Triage 33 content-loss residuals → B-17 epic OR Phase C resume | DONE Branch B — no single cause covered ≥50% of cluster, so no new B-18 epic filed. Posted 32-route grouping (C-1..C-6 carve-outs/clusters) to #666 for resumption. Analysis at `docs/migration-check-reports/2026-05-01-phase-c-input-post-b16-analysis.md` (commit 600057a). |
| #1124 | Re-baseline Phase E #668 with post-B-16 artifact diff | DONE — #668 body updated with new baseline numbers (sitemap.xml went from -68/+28 to -6/+15 — the headline improvement). |
| #1125 | Super-epic comment + close resume issue #1120 | DONE — super-epic #663 status comment posted, resume issue #1120 closed. |

### Dependent issue states

- #1119 (Phase B-16 epic) — **CLOSED** 2026-04-30
- #666 (Phase C mop-up) — **CLOSED** 2026-04-30 via PR #1329 (phase-c-mop-up landed C-1..C-6)
- #1120 (resume entry-point) — **CLOSED** 2026-04-30
- #1327 (B-17 comparator-mismatch sibling) — **CLOSED** 2026-04-30 via PR #1328

### Epic-PR

#1126 — **MERGED** 2026-04-30 into super-epic base `base/zfb-migration-parity`. Merge commit visible in `git log origin/base/zfb-migration-parity`. The mandatory "merge epic-PR into super-epic base" step is already satisfied; no further `/x-wt-teams` work remains for #1121.

### What still belongs to super-epic #663 (not this epic)

- Phase E #668 — re-baselined here, ready to run as its own `/x-wt-teams` session.
- Phase B #665 — parent placeholder; no longer a workstream.

Closing.

**Comment by Takazudo** [comment]
Workflow complete. Epic-PR #1126 merged into base/zfb-migration-parity on 2026-04-30. All sub-issues closed.
